### ISSUE TITLE
Speed limit is null always

### ISSUE THREAD

**Issue by SaeedNoshadi89** [mapbox/mapbox-navigation-android#6864]
I used LocationObserver and mapboxMachingResult to get some info like speed limit but I get null value for that what is the problem?

https://stackoverflow.com/questions/75199145/mapbox-speedlimit-return-null-value-in-android

**Comment by cafesilencio** [comment]
Mapbox doesn't have speed limit information for every road and every area.  Can you provide an example latitude and longitude for an origin and destination?  I'll see if I can reproduce the same results you reported.

**Comment by SaeedNoshadi89** [comment]
Sure, but as you know, according to the document, if data is not be exist observer returns "--" value.

You can see this points: 


listOf(
                Point.fromLngLat(51.359388, 35.773914),
                Point.fromLngLat(51.370464, 35.775621),
                Point.fromLngLat(51.375352, 35.776535),
            )

**Comment by cafesilencio** [comment]
The "--" will be displayed by the `SpeedLimitView` as indicated in the documentation here: https://docs.mapbox.com/android/navigation/guides/ui-components/speed-limit/.  I don't recall seeing this class in the source code you provided.  

Data from the `LocationObserver`, which may include a null value for `SpeedLimit`, goes to the `MapboxSpeedInfoApi` class to be processed.  The data returned can be used by your app. for displaying in your own view or it can (optionally) be rendered by the `SpeedLimitView` to display the data if you're using it.

I used one of our internal activities to test the coordinates you provided. [The source is available here](https://github.com/mapbox/mapbox-navigation-android/blob/09b52480c921d4acb03e206da778d4a4735a6477/qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/SpeedInfoActivity.kt).  I replaced the origin and destination in the `findRoute()` method with the first and last coordinates you provided.  I see the speed limit displayed.

May I suggest reviewing our example activity as a guide to get your app. working?

![Screenshot_20230124-154533](https://user-images.githubusercontent.com/2249818/214383168-eacef3ff-57c9-4907-9489-aaff2eae8dbc.png)

**Comment by SaeedNoshadi89** [comment]
I don't want to use any view to showing speed limit, I just need that data to record to csv file

**Comment by SaeedNoshadi89** [comment]
Can I get speed limit data without a route? for single point, not origin and destination. I want to get a speed limit data for current location. if yes please let me know how?

**Comment by cafesilencio** [comment]
No. Not every point is a routable point. The navigation system needs to process points as part of the road graph. There's no API to provide the speed limit for any given point.
